src/utils/crypto.ts

```
@@ -55,3 +57,43 @@ export const withEphemeralKey = async <R>(
   const ephemeralContext = createCryptoContextFor(keytag);
   return fn(ephemeralContext).finally(() => deleteKey(keytag));
 };
+/**
+ * Converts a certificate string to PEM format.
+ *
+ * @param certificate - The certificate string.
+ * @returns The PEM-formatted certificate.
+ */
+export const convertCertToPem = (certificate: string): string =>
+  `-----BEGIN CERTIFICATE-----\n${certificate}\n-----END CERTIFICATE-----`;
+
+/**
+ * Parses the public key from a PEM-formatted certificate.
+ *
+ * @param pemCert - The PEM-formatted certificate.
+ * @returns The public key object.
+ * @throws Will throw an error if the public key is unsupported.
+ */
+export const parsePublicKey = (
+  pemCert: string
+): RSAKey | KJUR.crypto.ECDSA | undefined => {
+  const x509 = new X509();
+  x509.readCertPEM(pemCert);
+  const publicKey = x509.getPublicKey();
+
+  if (publicKey instanceof RSAKey || publicKey instanceof KJUR.crypto.ECDSA) {
+    return publicKey;
+  }
+
+  return undefined;
+};
```

To avoid the dependency on jsrsasign what do you think about using a JWK and therefore [PublicKey type](https://github.com/pagopa/io-react-native-crypto/blob/fed9420e0bb961a410642dc13718518e5aeda917/src/index.tsx#L74C13-L74C22) from @pagopa/io-react-native-crypto ?

---

It can be used for this case, but `jsrsasign` cannot be removed for now, as it's still used in other scenarios, for example, in this file at line 80

---

It's fine to use it internally within the function. I would just avoid using types from other packages like return .

---

Ok, so I'll change it asap

---

What do you think about throwing an error instead of returning undefined?

---

Addressed [here](https://github.com/pagopa/io-react-native-wallet/pull/265/commits/6697ef63e63dda23e0b3b0971b7f210bb392ef1e). 
In the end, the `return undefined` was removed in favor of throwing an error if the `publicKey` is not supported.